### DISCUSSION TITLE
Invalid RSS feed of my FreshRSS account

### DISCUSSION THREAD

**Discussion by mizzunet** [FreshRSS/FreshRSS#5558] [Q&A] [answered]
* I have added some RSS from various sources in FreshRSS account. 
* I'd like to read feeds of my account in Feeder android app. 
* I copied  RSS feed URL and pasted into the app. But it failed to parse
![image](https://github.com/FreshRSS/FreshRSS/assets/10193999/0150cde8-4a68-4fd1-9c32-d8f8edc2968a)
* Later I found that the RSS is invalid

Here's URL: https://rss.hostux.net/i/?a=rss&get=c_15&rid=64cac1d703759&hours=168
Validation: https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Frss.hostux.net%2Fi%2F%3Fa%3Drss%26get%3Dc_15%26rid%3D64cac1d703759%26hours%3D168
`It looks like this is a web page, not a feed. I looked for a feed associated with this page, but couldn't find one. Please enter the address of your feed to validate.`
Is it a bug? Can I read FreshRSS feeds in any clients?

Edit:
* Just found that it is valid when I give direct input https://validator.w3.org/feed/#validate-by-input

Edit 2:
* Hm, the reason seems to be the feed URL is providing content-type `text/html`. It has to be `text/xml`
* example
```
❯ curl -i "https://rss.hostux.net/i/?a=rss&rid=64cac012b3ebe" | grep content-type
content-type: text/html; charset=UTF-8
```
```
❯ curl -i  "https://www.xda-developers.com/feed/" | grep content-type
 content-type: text/xml; charset=UTF-8
```

**Comment by Frenzie** [comment]
Most important here, what's the server type/configuration? Presumably of secondary importance here but still always good to know is the version (e.g., if it's dev then it might matter after all). 

For me the header looks like this:
```
Content-Type | application/rss+xml; charset=utf-8
```

**Reply by mizzunet** [reply]
> Most important here, what's the server type/configuration? 

Sorry, it's a public instance. I'm just a user. I'll contact the owner


Version 1.21.0

> Content-Type | application/rss+xml; charset=utf-8

You got this for the URL I given? Or for a a URL you taken from your instance?

**Reply by Frenzie** [reply]
For my own. That being said, I assumed you removed the token but are you sure your cURL isn't just getting the 403 page? :-)

That is, you'd normally need `&user=abc&token=etcetc`

**Reply by mizzunet** [reply]
No, that URL was of the RSS icon button shown in screenshot.

**Reply by Frenzie** [reply, marked as answer]
What if you check in settings → profile? Because in that case the validator and curl are definitely just getting the 403 page.

**Reply by mizzunet** [reply]
Exactly, thank you very much ❤️ 

* right, the authentication field was empty, so accessing RSS outside was disabled
* but still the URL displayed RSS XML of the feeds on the browser, probably was because the cookies.
* I just set a auth code. Now I can setup in Feeder.

Have a nice day :)
